internal/config/releases.go

```diff
@@ -0,0 +1,102 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/layer5io/meshery-adapter-library/adapter"
+)
+
+// cacheCheck is the time in weeks when the releases
+// should be updated grom github release pages
+const cacheCheck = 1
+
+// Release is used to save the release informations
+type Release struct {
+	ID      int             `json:"id,omitempty"`
+	TagName string          `json:"tag_name,omitempty"`
+	Name    adapter.Version `json:"name,omitempty"`
+	Draft   bool            `json:"draft,omitempty"`
+	Assets  []*Asset        `json:"assets,omitempty"`
+}
+
+// Asset describes the github release asset object
+type Asset struct {
+	Name        string `json:"name,omitempty"`
+	State       string `json:"state,omitempty"`
+	DownloadURL string `json:"browser_download_url,omitempty"`
+}
+
+// getLatestReleaseNames returns the names of the latest releases
+// limited by the "limit" parameter. The first version in the list
+// is always is the latest "stable" version.
+func getLatestReleaseNames(limit int) ([]adapter.Version, error) {
+	releases, err := GetLatestReleases(10)
```

Comment:
Even though top 10, you may not find any stable version of `linkerd2-ctl`  as we can see the linkerd2-ctl -edge and stable released on the same page. There may need another way.

Reply:
@Aisuko yeah you are right but I don't think github api supports such filtering. I can only think of two other (pretty inefficient) approaches:
1. We query for 100 (max allowed by the api) latest releases and then filter out top 10 or something like that.
2. Query for 10 releases and keep on querying until we find at least one stable release.

Any suggestions?

Reply:
Yep, I hit the same issue, the latest one stable version almost 17, and if I remember right I imported the linkerd2 source code and render the manifest. Actually, this is not a good idea to solve our current situation. We should keep the stable version, may we config the latest stable version to the configuration?

Reply:
hmm are right @Aisuko , We might need to match some regex on the versions we get to look for just the stable ones.

Reply:
@kumarabd, the issue is that https://api.github.com/repos/linkerd/linkerd2/releases may not return the "stable" releases at all because linkerd places "edge" releases and "stable" releases on same page, so the entire response body may contain only "edge" releases. 
The function invocation that I have written at the moment kind of assumes that out of 10 releases that the api returns, one of them will at least be a "stable" release.